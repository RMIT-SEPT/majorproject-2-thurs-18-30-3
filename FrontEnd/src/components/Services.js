import React, { Component } from 'react';
import { Table, Button } from 'reactstrap';
import { Link } from 'react-router-dom';

import '../containers/App.css';

//Displays list of Employees 

class Services extends Component {
    constructor(props){
        super(props);
    }

	//Props passed down by ServiceListContainer

    render() {
		let services = this.props.services.map((service) => {
			return (
				//Render an item in booking list for each booking fetched
				<tr key ={service.id}>

					<td>{service.id}</td>

					<Link to={`/services/${service.id}`} className = "booking-link">
						<td>{service.title}</td>
					</Link>

					<td>{service.body}</td>

					<td>
						<Button color = "success" size = 'l'>Book</Button>
					</td>
				</tr>
				
			) 
		});
		return (
		<div className = "services-list">
			<Table>
				<thead>
					<tr>
						<th>#</th>
						<th>SERVICE</th>
						<th>DESCRIPTION</th>
						<th>ACTION</th>
					</tr>
				</thead>

				<tbody>
					{services}
				</tbody>
			</Table>
		</div>
		);
	}

}
export default Services;