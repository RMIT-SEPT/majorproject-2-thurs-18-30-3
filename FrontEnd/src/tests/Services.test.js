import React from 'react'

import Services from '../components/Services'
import {StaticRouter} from 'react-router'
import renderer from 'react-test-renderer'
import moment from 'moment'
import {render, screen, fireEvent, queryByText} from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

const dummyService = {id: 1, name: 'test', description: 'this is a test', photo: "iPhone12.jpg"}
const searchFuncMock = jest.fn()

const setup = () => {
  return render(
    <StaticRouter>
      <Services services={[dummyService]} searchFunc={searchFuncMock} />
    </StaticRouter>
  )
}

//snapshot test
test('Test Page Renders', () => {
  const comp = renderer
    .create(
      <StaticRouter>
        <Services services={[dummyService]} searchFunc={searchFuncMock} />
      </StaticRouter>
    )
    .toJSON()
  expect(comp).toMatchSnapshot()
})

test('Test Page Contents', () => {
  const container = setup()
  expect(screen.getByRole('heading')).toHaveTextContent('Services')
  expect(screen.getByRole('searchbox'))
  expect(screen.getByRole('main')).toHaveClass('services-gallery')
})

//Ensure gallery is displaying props
test('Check Gallery', () => {
  const container = setup()
  const gallery = screen.getByRole('main')

  expect(gallery).toHaveTextContent('this is a test')
})


//Tests for the search field at the top of the gallery
test('Test Search PlaceHolder', () => {
  const container = setup()
  const input = screen.getByRole('searchbox')

  expect(input).toHaveAttribute('placeholder', 'Search')
})

test('Test Search Input', () => {
  const container = setup()
  const input = screen.getByRole('searchbox')
  fireEvent.change(input, {target: {value: 'test'}})
  expect(input.value).toBe('test')
})

test('Test Search Fires', () => {
  const container = setup()
  const input = screen.getByRole('searchbox')
  fireEvent.change(input, {target: {value: 'test'}})

  expect(searchFuncMock).toHaveBeenCalled()
})
