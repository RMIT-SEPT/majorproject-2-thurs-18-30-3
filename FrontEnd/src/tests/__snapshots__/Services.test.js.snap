// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page Renders 1`] = `
<div>
  <h1>
    Services
  </h1>
  <div
    className="sector-heading"
  >
    <form
      data-testid="datePickerA"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
        role="listbox"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="date"
          id="date-label"
        >
          Display Services From
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            defaultValue="2020-08-28"
            disabled={false}
            id="date"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
        </div>
      </div>
    </form>
    <span>
        &gt; 
    </span>
    <form
      data-testid="datePickerB"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
        role="listbox"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="date"
          id="date-label"
        >
          Display Services To
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            defaultValue="2020-08-28"
            disabled={false}
            id="date"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
        </div>
      </div>
    </form>
    <div
      className="dummy"
    />
    <form>
      <input
        placeholder="Search"
        role="searchbox"
      />
    </form>
  </div>
  <hr
    className="sector-divider"
  />
  <div
    className="services-gallery"
    role="main"
  >
    <a
      href="/services/1"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        aria-label={null}
        className="service-card btn btn-secondary"
        onClick={[Function]}
        role="cell"
      >
        <img
          alt="{service.name}"
          src="not-found.png"
        />
        <div
          className="card-content"
        >
          <span />
          <hr />
          <p>
            this is a test
          </p>
        </div>
      </button>
    </a>
  </div>
</div>
`;
